### PR TITLE
Add timestamps to debug logs.

### PR DESCRIPTION
**Pull request by fli**
Prefixes each log line with a Unix-style timestamp.

### CODE REVIEW

**Comment by fli** [comment]
How's this look? Lines are in format
```
year-month-day hours:minutes:seconds.milliseconds debugging-line
```

File: src/chess/uciloop.cc [hunk 1]
```diff
@@ -215,6 +228,13 @@ void UciLoop::SetLogFilename(const std::string& filename) {
   }
 }
 
+void UciLoop::WriteDebugLogLine(const std::string& line) {
+  static std::mutex logging_mutex;
+  std::lock_guard<std::mutex> lock(logging_mutex);
+  if (debug_log_)
+    debug_log_ << GetTimestamp() << ' ' << line << std::endl << std::flush;
```
Comment (mooskagh) [inline src/chess/uciloop.cc]: `std::endl` does `std::flush` itself (unlike outputting '\n'), so no need to flush explicitly.

File: src/chess/uciloop.cc [hunk 2]
```diff
@@ -122,13 +122,26 @@ bool ContainsKey(const std::unordered_map<std::string, std::string>& params,
   return params.find(key) != params.end();
 }
 
+std::string GetTimestamp() {
+  using namespace std::chrono;
+  auto time_now = system_clock::now();
+  auto ms =
+      duration_cast<milliseconds>(time_now.time_since_epoch()).count() % 1000;
+  auto timer = system_clock::to_time_t(time_now);
+  std::tm bt = *std::localtime(&timer);
```
Comment (mooskagh) [inline src/chess/uciloop.cc]: Maybe feed `std::localtime` directly to `put_time` instead of creating a temporary copy of the structure?

**Comment by fli** [comment]
Thanks, made changes.

**Pull request merged**